_source/_posts/2022-07-06-spa-web-security.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: blog_post
+title: "Defend Your SPA from Security Woes"
+author: alisa-duncan
+by: advocate
+communities: [security,javascript]
+description: "Learn the basics of web security and how to apply web security foundation to protect your Single Page Applications."
+tags: [security, react, vue, angular]
+tweets:
+- "Protect your SPA and web 🍪  by keeping these web security tips in mind!"
+- "Confused about web security? 🤔 Read this post to demystify actionable steps you can take to protect your SPA. ✅"
+image: blog/spa-web-security/social.jpg
+type: awareness
+---
+
+There's a lot of information floating out there about web security. But when I read through the material, I noticed some information wasn't up to date, or they were written specifically for traditional server-rendered web applications, or they recommended anti-patterns. In a series of posts, I will cover web security concerns that all web devs should be aware of, emphasizing client-side applications, namely Single Page Applications (SPA). Furthermore, I'm not going to get into the nitty-gritty of web security, cryptography, and networking, and instead, focus on the main takeaways and high-level to-do lists. I'll also provide links where you can dive in deeper.
+
+<!-- Table with all the posts here as each post publishes -->
+
+So why do we worry about web security anyway? When we have vulnerabilities in our applications, it's an avenue for exploitation. In the wrong hands, exploiting the vulnerabilities can cause risks to your application, data, reputation, and bottom line. In the words of my coworker [Aaron Parecki](https://developer.okta.com/blog/authors/aaron-parecki/), it all boils down to one thing—liability. And we don't want to expose ourselves, our applications, and our companies to liability if we can help it. If you feel threatened, it's a good time to read on and identify how to harden your application.
```

Comment:
```suggestion
So why do we worry about web security anyway? When we have vulnerabilities in our applications, it's an avenue for exploitation. In the wrong hands, exploiting the vulnerabilities can cause risks to your application, data, reputation, and bottom line. In the words of my coworker [Aaron Parecki](https://developer.okta.com/blog/authors/aaron-parecki/), it all boils down to one thing—liability. And we don't want to expose ourselves, our applications, and our companies to liability. If you feel threatened, it's a good time to read on and identify how to harden your application.
```
I think "if we can help it" waters down the oomph of your statement, unless I'm missing the point of qualifying here,,,